res/controllers/Pioneer-DDJ-SB-scripts.js

```diff
@@ -898,88 +840,81 @@ PioneerDDJSB.rotarySelectorShiftedClick = function(channel, control, value, _sta
-};
-
-PioneerDDJSB.fxKnobShiftedLSB = function(channel, control, value, status) {
-    var deck = channel - 4,
-        fullValue = (PioneerDDJSB.fxKnobShiftedMSB[deck] << 7) + value;
-
-    if (PioneerDDJSB.softTakeoverEmulation(deck, 4, PioneerDDJSB.fxKnobShiftedMSB[deck])) {
-        engine.setValue('[EffectRack1_EffectUnit' + (deck + 1) + ']', 'super1', fullValue / 0x3FFF);
-    }
-};
-
-PioneerDDJSB.fxKnobMSB = function(channel, control, value, status) {
-    PioneerDDJSB.fxKnobMSB[channel - 4] = value;
-};
-
-PioneerDDJSB.fxKnobLSB = function(channel, control, value, status) {
-    var deck = channel - 4,
-        anyButtonPressed = false,
-        fullValue = (PioneerDDJSB.fxKnobMSB[deck] << 7) + value,
-        parameter;
-
-    for (parameter = 0; parameter < 3; parameter++) {
-        if (PioneerDDJSB.fxButtonPressed[deck][parameter]) {
-            anyButtonPressed = true;
-        }
-    }
-
-    if (!anyButtonPressed) {
-        if (PioneerDDJSB.softTakeoverEmulation(deck, 3, PioneerDDJSB.fxKnobMSB[deck])) {
-            engine.setValue('[EffectRack1_EffectUnit' + (deck + 1) + ']', 'mix', fullValue / 0x3FFF);
-        }
-    } else {
-        for (parameter = 0; parameter < 3; parameter++) {
-            if (PioneerDDJSB.fxButtonPressed[deck][parameter] && PioneerDDJSB.softTakeoverEmulation(deck, parameter, PioneerDDJSB.fxKnobMSB[deck])) {
-                engine.setParameter(
-                    '[EffectRack1_EffectUnit' + (deck + 1) + '_Effect1]',
-                    'parameter' + (parameter + 1),
-                    fullValue / 0x3FFF
-                );
+        components.Button.call(this);
+    };
+    this.EffectButton.prototype = new components.Button({
+        input: function(channel, control, value, status) {
+            if (this.isPress(channel, control, value, status)) {
+                this.isLongPressed = false;
+                this.longPressTimer = engine.beginTimer(this.longPressTimeout, function() {
+                    var effectGroup = "[EffectRack1_EffectUnit" + unitNumber + "_Effect" + this.buttonNumber + "]";
+                    script.toggleControl(effectGroup, "enabled");
+                    this.isLongPressed = true;
+                }, true);
+            } else {
+                if (!this.isLongPressed) {
+                    var focusedEffect = engine.getValue(eu.group, "focused_effect");
+                    if (focusedEffect === this.buttonNumber) {
+                        engine.setValue(eu.group, "focused_effect", 0);
+                    } else {
+                        engine.setValue(eu.group, "focused_effect", this.buttonNumber);
+                    }
+                }
+                this.isLongPressed = false;
+                engine.stopTimer(this.longPressTimer);
             }
+        },
+        outKey: "focused_effect",
+        output: function(value, _group, _control) {
+            this.send((value === this.buttonNumber) ? this.on : this.off);
+        },
```

Comment:
fyi, this could be handled more elegantly via outValueScale:
```suggestion
        outValueScale: function(value) {
            return (value === this.buttonNumber) ? this.on : this.off;
        },
```

Reply:
Solved in https://github.com/mixxxdj/mixxx/pull/12280/commits/0406cbb5a95fe959399f29c22fe8a974229b6cad